src/components/EventList.js

```diff
@@ -23,16 +27,42 @@ const EventList = ({
     contentContainerStyle={styles.container}
     data={events}
     keyExtractor={event => event.sys.id}
-    renderItem={({ item: event }) => (
-      <View style={styles.eventListItem}>
-        <TouchableOpacity
-          delayPressIn={50}
-          onPress={() => onPress(event.sys.id)}
-        >
-          <EventCard name={event.fields.name[locale]} />
-        </TouchableOpacity>
-      </View>
-    )}
+    renderItem={({ item: event }) => {
+      const startTime = removeTimezoneFromDateString(
+        event.fields.startTime[locale]
+      );
+      const endTime = removeTimezoneFromDateString(
+        event.fields.endTime[locale]
+      );
+      const dateFormat = "DD MMMM YYYY";
+      const timeFormat = "HH:mm";
+      const dateDisplay = isSameDay(startTime, endTime)
+        ? formatDate(startTime, dateFormat)
+        : `${formatDate(startTime, dateFormat)} - ${formatDate(
+            endTime,
+            dateFormat
+          )}`;
+      const timeDisplay = `${formatDate(startTime, timeFormat)} - ${formatDate(
+        endTime,
+        timeFormat
+      )}`;
+      return (
+        <View style={styles.eventListItem}>
+          <TouchableOpacity
+            delayPressIn={50}
+            onPress={() => onPress(event.sys.id)}
+          >
+            <EventCard
+              date={dateDisplay}
```

Comment:
Just a reminder of the conversation from yesterday: This prop should get replaced with two separate props: `startTime` and `endTime` which you can pass in the same way you do `name`. That change will allow you to move all of the date formatting logic into `EventCard` and make `renderItem` much simpler again.

The component got a whole lot more complicated in #62, so it will help a lot not to give it any more jobs.